addon/components/-private/base-pop-menu.js

```diff
@@ -251,9 +271,58 @@ export default class BasePopMenuComponent extends Component {
 
     } else if (closest(target, '[data-close]:not([disabled])')) {
       // We still want to allow purposeful closing of the popover from within,
-      // so this closes the popover if the click target has [data-popover-close] attribute
+      // so this closes the popover if the click target has [data-close] attribute
 
       this._removePopover();
     }
   };
+
+  /**
+   * When focus is on the trigger, determine specific key actions
+   */
+  _triggerKeyHandler = (event) => {
+    const keyCode = event.code;
+
+    // Pressing Enter or Space opens the popper
+    if (
+      (keyCode == 'Enter' || keyCode == 'Space')
+      && !this.get('isOpen')
+      ) {
+      this._openPopoverHandler();
+    }
+
+    // Close on esc if we never entered the popper
+    // (Enter is the same as re-clicking the button)
+    else if (
```

Comment:
I also think we prevent you from breaking on `else` like this, it should be on the same line as the last curly